### PR TITLE
refactor(browser): use client sdk

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

`@logto/browser` is now based on `@logto/client`.

## Testing

UTs.

### CODE REVIEW

File: packages/browser/src/index.ts [hunk 1]
```diff
@@ -1,377 +1,76 @@
...
-  public isSignInRedirected(url: string): boolean {
-    const { signInSession } = this;
-
-    if (!signInSession) {
-      return false;
+  setItem(key: 'idToken' | 'refreshToken' | 'accessToken' | 'signInSession', value: string): void {
+    if (key === 'signInSession') {
+      sessionStorage.setItem(this.storageKey, value);
+    } else {
+      localStorage.setItem(`${this.storageKey}:${key}`, value);
```
Comment (gao-sun) [inline packages/browser/src/index.ts:55]: better to use return early or switch with deafult?

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -11,16 +11,17 @@ importers:
       lerna: ^5.0.0
       typescript: ^4.5.5
     devDependencies:
-      '@commitlint/cli': 17.0.1
-      '@commitlint/config-conventional': 17.0.0
+      '@commitlint/cli': 17.0.3
```
Comment (gao-sun) [inline pnpm-lock.yaml:14]: deps upgrade should be a separate PR

**Review by gao-sun — APPROVED** [review]

File: pnpm-lock.yaml (hunk 2 above)
Comment (wangsijie) [reply pnpm-lock.yaml]: Oh, removed

File: packages/browser/src/index.ts (hunk 1 above)
Comment (wangsijie) [reply packages/browser/src/index.ts:55]: updated

File: packages/browser/package.json [hunk 3]
```diff
@@ -29,23 +29,18 @@
...
-    "jose": "^4.5.0",
-    "lodash.get": "^4.4.2",
-    "lodash.once": "^4.1.1",
-    "superstruct": "^0.16.0"
+    "@logto/client": "^1.0.0-alpha.2",
+    "@silverhand/essentials": "^1.1.6"
   },
   "devDependencies": {
     "@jest/types": "^27.5.1",
+    "@logto/js": "^1.0.0-alpha.2",
```
Comment (demonzoo) [inline packages/browser/package.json:37]: why do we need to add `@logto/js` as dev dependency?

File: packages/client/src/index.ts [hunk 4]
```diff
@@ -26,9 +26,10 @@ import { LogtoClientError } from './errors';
 import { Storage } from './storage';
 import { buildAccessTokenKey, getDiscoveryEndpoint } from './utils';
 
-export type { IdTokenClaims, LogtoErrorCode } from '@logto/js';
-export { LogtoError, OidcError, Prompt } from '@logto/js';
+export type { IdTokenClaims, LogtoErrorCode, LogtoRequestErrorBody } from '@logto/js';
```
Comment (demonzoo) [inline packages/client/src/index.ts:29]: can we export these types from `@logto/client`?

File: packages/browser/package.json (hunk 3 above)
Comment (wangsijie) [reply packages/browser/package.json:37]: Used for mockups, make it easy to test `storage`

File: packages/browser/src/index.ts [hunk 5]
```diff
@@ -1,377 +1,76 @@
...
-  generateState,
-  IdTokenClaims,
-  Prompt,
-  Requester,
-  revoke,
-  verifyAndParseCodeFromCallbackUri,
-  verifyIdToken,
-  withReservedScopes,
-} from '@logto/js';
+import BasicClient, {
```
Comment (demonzoo) [inline packages/browser/src/index.ts:1]: nit: Basic -> Base
`BaseClient` sounds better to me. Feels like something you should extend before use. But `BasicClient` sounds like a minimal usable client.

File: packages/browser/src/index.ts [hunk 6]
```diff
@@ -1,377 +1,76 @@
...
 
-    return token;
+  constructor(appId: string) {
+    this.storageKey = `${logtoStorageItemKeyPrefix}:${appId}`;
   }
 
-  public getIdTokenClaims(): IdTokenClaims {
-    if (!this.idToken) {
-      throw new LogtoClientError('not_authenticated');
+  getItem(key: 'idToken' | 'refreshToken' | 'accessToken' | 'signInSession'): Nullable<string> {
```
Comment (demonzoo) [inline packages/browser/src/index.ts:43]: nit: The type of the storage key can be exported from `@logto/client`

Comment (wangsijie) [reply packages/browser/src/index.ts]: Aha, that is written by coplit, updated

File: packages/browser/src/index.ts (hunk 5 above)
Comment (wangsijie) [reply packages/browser/src/index.ts]: updated

File: packages/browser/package.json (hunk 3 above)
Comment (wangsijie) [reply packages/browser/package.json:37]: created LOG-3636 to refactor in the future.

File: packages/browser/src/index.ts (hunk 6 above)
Comment (demonzoo) [reply packages/browser/src/index.ts]: Finally, I find myself better than AI. lol

File: packages/browser/src/index.test.ts [hunk 7]
```diff
@@ -86,346 +66,39 @@ jest.mock('jose', () => ({
...
-
-      const logtoClient = new LogtoClient({ endpoint, appId }, requester);
-      const [accessToken_1, accessToken_2] = await Promise.all([
-        logtoClient.getAccessToken(),
-        logtoClient.getAccessToken(),
-      ]);
-
-      expect(requester).toHaveBeenCalledWith(tokenEndpoint, expect.anything());
-      expect(accessToken_1).toEqual('access_token_value');
-      expect(accessToken_2).toEqual('access_token_value');
```
Comment (demonzoo) [inline packages/browser/src/index.test.ts:351]: Why do we remove these accessToken related test cases? Are they covered in `@logto/client`?

**Review by demonzoo — APPROVED** [review]
Approved with comments. Please make sure we don't lose test cases from this code refactor

File: packages/browser/src/index.test.ts (hunk 7 above)
Comment (wangsijie) [reply packages/browser/src/index.test.ts:351]: The answer is YES

Comment (wangsijie) [reply packages/browser/src/index.test.ts:351]: Besides, in this refactor, all test cases are remained.
